test/Npgsql.Tests/Types/JsonPathTests.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Threading.Tasks;
+using NpgsqlTypes;
+using NUnit.Framework;
+using static Npgsql.Tests.TestUtil;
+
+namespace Npgsql.Tests.Types
+{
+    public class JsonPathTests : MultiplexingTestBase
+    {
+        public JsonPathTests(MultiplexingMode multiplexingMode)
+            : base(multiplexingMode) { }
+
+        static readonly object[] ReadWriteCases = new[]
+        {
+            new object[] { "'$'::text", "$" },
+            new object[] { "'$\"varname\"'::text", "$\"varname\"" },
+        };
+
+        [Test]
+        [TestCaseSource(nameof(ReadWriteCases))]
+        public async Task Read(string query, string expected)
+        {
+            using var conn = await OpenConnectionAsync();
+            MinimumPgVersion(conn, "12.0", "The jsonpath type was introduced in PostgreSQL 12");
+
+            using var cmd = new NpgsqlCommand("SELECT " + query, conn);
+            using var rdr = await cmd.ExecuteReaderAsync();
+
+            rdr.Read();
+            Assert.That(rdr.GetFieldValue<string>(0), Is.EqualTo(expected));
+            Assert.That(rdr.GetTextReader(0).ReadToEnd(), Is.EqualTo(expected));
+        }
+
+        [Test]
+        [TestCaseSource(nameof(ReadWriteCases))]
+        public async Task Write(string query, string expected)
+        {
+            using var conn = await OpenConnectionAsync();
+            MinimumPgVersion(conn, "12.0", "The jsonpath type was introduced in PostgreSQL 12");
+
+            using var cmd = new NpgsqlCommand("SELECT @p, @p::text = " + query, conn) { Parameters = { new NpgsqlParameter("p", NpgsqlDbType.JsonPath) { Value = expected } } };
+            using var rdr = await cmd.ExecuteReaderAsync();
+
+            rdr.Read();
+            Assert.That(rdr.GetFieldValue<string>(0), Is.EqualTo(expected));
+            Assert.That(rdr.GetFieldValue<bool>(1));
```

Comment:
Like the read test above, I'm not sure what `@p::text = " + query` tests exactly, beyond what the simple `@p` does... We send the jsonpath parameter @p to PG (just like the first `@p` does), make PG convert it to a text and compare to another text... Unless I'm missing something just `SELECT @p` is enough, no?